File: TicTacToe2D/tictactoe2d.py

BOARD = [["EMPTY","EMPTY","EMPTY"],["EMPTY","EMPTY","EMPTY"],["EMPTY","EMPTY","EMPTY"]];


def printGameBoard(BOARD):
	for(int count = 0 ; count< BOARD.length ; count++ ){
		print("%s | %s | %s \n " %(BOARD[count][0],BOARD[count][1],BOARD[count][2]));
		if(count<2){
			print("--------------------");


def playerInput(BOARD):
	playerInput =int(input("Enter row number and column numberyy \ne.g 12 , 13\n  : "));
	if playerInput >= 11 && playerInput <= 13 && BOARD[(playerInput/10)-1][(playerInput%10)-1] == "EMPTY" :
		BOARD[(playerInput/10)-1][(playerInput%10)-1] = firstPlayer;

	elif(playerInput >= 21 && playerInput <= 23 && BOARD[(playerInput/10)-1][(playerInput%10)-1] == "EMPTY":
		BOARD[(playerInput/10)-1][(playerInput%10)-1] = firstPlayer;

	elif(playerInput >= 31 && playerInput <= 33 && BOARD[(playerInput/10)-1][(playerInput%10)-1] == "EMPTY":
		BOARD[(playerInput/10)-1][(playerInput%10)-1] = firstPlayer;

	else:
		print(" Space Occupied ");

def checkHorizontal(BOARD):
	if BOARD[0][0] == BOARD[0][1] && BOARD[0][0]  == BOARD[0][2] && BOARD[0][0] != "EMPTY":
		winner = BOARD[0][0];
		return true ;
		
	elif BOARD[1][0] == BOARD[1][1] && BOARD[1][0]  == BOARD[1][2] && BOARD[1][0] != "EMPTY":
		winner = BOARD[1][0];
		return true ;
		
	elif(BOARD[2][0] == BOARD[2][1] && BOARD[2][0]  == BOARD[2][2] && BOARD[2][0] != "EMPTY":
			winner = BOARD[2][0];
			return true ;
		return false;

def checkvertical(board):
	global winner = " ";
	if BOARD[0][0] == BOARD[1][0] && BOARD[0][0]  == BOARD[2][0] && BOARD[0][0] != "EMPTY":
		winner = BOARD[0][0];
		return true ;

	elif BOARD[0][1] == BOARD[1][1] && BOARD[0][1]  == BOARD[2][1] && BOARD[0][1] != "EMPTY":
			winner = board[0][1];
			return true ;
	elif(BOARD[0][2] == BOARD[1][2] && BOARD[0][2]  == BOARD[2][2] && BOARD[0][2] != "EMPTY":
			winner = BOARD[0][2];
			return true ;
	return false;
	
def checkDiagonals(board):
	if BOARD[0][0] == BOARD[1][1] && BOARD[0][0]  == BOARD[2][2] && BOARD[0][0] != "EMPTY" :
		winner = BOARD[0][0];
		return true ;
	elif BOARD[2][0] == BOARD[1][1] && BOARD[2][0]  == BOARD[0][2] && BOARD[2][0] != "EMPTY":
		winner = BOARD[2][0];
		return true ;
		
	return false;

def checkTie(String[][] BOARD):
	global gameisRunning = true;
	for int index = 0; index < BOARD.length ; index++:
		for int count = 0; count < BOARD.length ; count++:
			if BOARD[index][count] == ("EMPTY") :
				return;
	print("Draw");
	gameisRunning = false;
		
def switchPlayer():
	global firstPlayer = "X";
	if firstPlayer == "X" :
		firstPlayer = "O";
	else:
		firstPlayer = "X" ;
def checkWin():
	if checkvertical(BOARD) or checkDiagonals(BOARD) or checkHorizontal(BOARD):
		print(f"the winner is {winner}");		
	

while(gameisRunning){
	printGameBoard(BOARD);
	playerInput(BOARD);
	checkWin();
	checkTie(BOARD);
	switchPlayer();

File: TicTacToe2D/TicTacToe2D.java
import java.util.Scanner;
public class TicTacToe2D{
	static Scanner input = new Scanner(System.in);
	static String[][] BOARD = {
				   {"EMPTY","EMPTY","EMPTY"},
				   {"EMPTY","EMPTY","EMPTY"},
				   {"EMPTY","EMPTY","EMPTY"}
				   };
	static String firstPlayer = "X";
	static String winner = " ";
	static Boolean gameisRunning = true;

	public static void main(String[] args){


		
		
		while(gameisRunning){

			printGameBoard(BOARD);
			playerInput(BOARD);
			checkWin();
			checkTie(BOARD);
			switchPlayer();
		}


	}
	public static void printGameBoard(String[][] BOARD){
		for(int count = 0 ; count< BOARD.length ){
		System.out.printf("%s | %s | %s \n ",BOARD[count][1],BOARD[count][2],BOARD[count][3]);
			if(count<=2){
				System.out.println("---------");
			}
		}
			
	}

	public static void playerInput(String[][] BOARD){
		System.out.print("Enter row number and column numberyy \ne.g 12 , 13\n  : ");
		int playerInput = input.nextInt();

		if(playerInput >= 11 && playerInput <= 13 && BOARD[playerInput-1] == "EMPTY" ){
			BOARD[playerInput-1/10][playerInput-1%10] = firstPlayer;
		}
		if(playerInput >= 21 && playerInput <= 23 && BOARD[playerInput-1] == "EMPTY" ){
			BOARD[playerInput-1/10][playerInput-1%10] = firstPlayer;
		}
		if(playerInput >= 31 && playerInput <= 33 && BOARD[playerInput-1] == "EMPTY" ){
			BOARD[playerInput-1/10][playerInput-1%10] = firstPlayer;
		}

		else {
			System.out.println(" Space Occupied ");
		}
		
	}
	
	public static boolean checkHorizontal(String[][] BOARD){
		if(BOARD[0] == BOARD[1] && BOARD[0]  == BOARD[2] && BOARD[0] != "EMPTY"){
			winner = board[2];
			return true ;
		}
		else if(BOARD[3] == BOARD[4] && BOARD[3]  == BOARD[5] && BOARD[3] != "EMPTY"){
			winner = board[2];
			return true ;
		}
		else if(BOARD[6] == BOARD[7] && BOARD[6]  == BOARD[8] && BOARD[6] != "EMPTY"){
			winner = BOARD[2];
			return true ;
		}

		return false;
	}


	public static boolean checkvertical(String[][] board){
		if(BOARD[0] == BOARD[3] && board[0]  == board[6] && board[0] != "EMPTY"){
			winner = board[0];
			return true ;
		}
		else if(board[1] == board[4] && board[1]  == board[7] && board[1] != "EMPTY"){
			winner = board[1];
			return true ;
		}
		else if(board[2] == board[5] && board[2]  == board[8] && board[2] != "EMPTY"){
			winner = board[2];
			return true ;
		}
		return false;
	}
	
	public static boolean checkDiagonals(String[][] board){
	
		if(board[0] == board[4] && board[0]  == board[8] && board[0] != "EMPTY"){
			winner = board[0];
			return true ;
		}
		else if(board[2] == board[4] && board[1]  == board[7] && board[1] != "EMPTY"){
			winner = board[2];
			return true ;
		}
		
		return false;
	}

	public static void checkTie(String[][] board){
		for(int count = 0; count < board.length ; count++){
			if (board[count].equals("EMPTY") ){
				
				return;
			}
		}
		System.out.println("Draw");
		gameisRunning = false;
		
		
	}



	public static void switchPlayer(){
	
		if(firstPlayer == "X"){
			firstPlayer = "O";
		}
		else{
			firstPlayer = "X" ;
		}
	}

	public static void checkWin(){
		if (checkvertical(board) || checkDiagonals(board) || checkHorizontal(board)){
			System.out.printf("the winner is %s%n",winner);		
	
		}
	}
	






}